### ISSUE TITLE
ValueError: numpy.ndarray size changed, may indicate binary incompatibility.

### ISSUE THREAD

**Issue by chudur-budur** [scikit-tda/ripser.py#128]
When I try to import `ripser`, I'm getting this error.

```
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject
```

```
Python version: 3.8.5
numpy: 1.19.2
ripser: 0.6.0
```

I tried with different combinations of `python` + `numpy` + `ripser` versions. But couldn't get it to work.

What is the correct `python` + `numpy` version for ripser to work?

**Comment by bdice** [comment]
My guess is that the published packages may need to be built using the oldest supported NumPy version in order to resolve this. You may be able to make this work by updating to the latest NumPy version (or something newer than whichever version was used to build the last release of ripser).

https://github.com/scikit-tda/ripser.py/blob/9d9b7c467b01e9d85f996cdc1cbefef9fe3613f5/.github/workflows/python-publish.yml#L29

https://pypi.org/project/oldest-supported-numpy/

For freud, a project I maintain that also uses Cython and builds against NumPy, we use this definition in the `pyproject.toml` build requirements for PEP 517/518 compatible builds: https://github.com/glotzerlab/freud/blob/master/pyproject.toml

**Comment by chudur-budur** [comment]
> My guess is that the published packages may need to be built using the oldest supported NumPy version in order to resolve this. You may be able to make this work by updating to the latest NumPy version (or something newer than whichever version was used to build the last release of ripser).
> 
> This line in the package publishing workflow may need to be changed to use `oldest-supported-numpy`.
> https://github.com/scikit-tda/ripser.py/blob/9d9b7c467b01e9d85f996cdc1cbefef9fe3613f5/.github/workflows/python-publish.yml#L29
> 
> https://pypi.org/project/oldest-supported-numpy/
> 
> For freud, a project I maintain that also uses Cython and builds against NumPy, we use this definition in the `pyproject.toml` build requirements for PEP 517/518 compatible builds: https://github.com/glotzerlab/freud/blob/master/pyproject.toml

Which numpy version is the oldest supported version for ripser? Where do I find that info?

**Comment by bdice** [comment]
@chudur-budur Try updating your NumPy version to the newest available (1.21.2 at the time of writing).

The comments about `oldest-supported-numpy` are about how to solve this problem by fixing the ripser build system so that users do not need to use the newest version of NumPy.

**Comment by bdice** [comment]
Accidentally closed the issue, wrong button. Sorry. 😬

**Comment by bdice** [comment]
I created #129 to attempt to solve this problem.

**Comment by chudur-budur** [comment]
> @chudur-budur Try updating your NumPy version to the newest available (1.21.2 at the time of writing). The version you use must be newer than the version used to build ripser, which is ... whatever version was available at the time that the ripser package was last released. It's probably not information that is easy to find.
> 
> The comments about `oldest-supported-numpy` are about how to solve this problem by fixing the ripser build system so that users do not need to use the newest version of NumPy.

Yes, `ripser==0.6.1` works with `numpy==1.21.1`. But I also need `tensorflow`, the version I'm using is `2.6.0`, which only works with `numpy==1.19.5`. But with `numpy==1.19.5`, ripser throws the same error. 

In that case what should I do? 

I have installed `ripser==0.6.0` to see if it works with `numpy==1.19.5`, but it doesn't. Same error comes up when I do `import ripser`.

**Comment by bdice** [comment]
@chudur-budur You may also build the package from source, instead of using a wheel from PyPI. You may need to use the pull request I created or disable pip build isolation to ensure that you build against a compatible NumPy version. Try cloning the repository and run `pip install . --no-use-pep517` from the repository root directory to build with isolation disabled.
